File: cpp/09proc/Proc37.cpp
#include <iostream>
#include <cmath>
using namespace std;

double Power1(double A, double B);

int main()
{
	//Task("Proc37");
	double P, number;
	cin >> P;
	for (int i = 1; i < 4; i++)
	{
		cin >> number;
		cout << Power1(number, P);
	}
	
	return 0;
}

double Power1(double A, double B)
{
	if (A <= 0)
		return 0;
	return exp(B * log(A));
}

File: php/09proc/proc37.php
<?php
function Power1($A, $B){
        $nat = 0;
        if($A > 0)
                $nat = exp($B * log($A));
        return $nat;
}
$A = mt_rand(-20, 20);
$B = mt_rand(-20, 20);
$C = mt_rand(-20, 20);
$P = mt_rand(-10, 10);
echo "A = ".$A;
echo "<br>B = ".$B;
echo "<br>C = ".$C;
echo "<br>P = ".$P;
echo "<br>A^P = ".Power1($A, $P);
echo "<br>B^P = ".Power1($B, $P);
echo "<br>C^P = ".Power1($C, $P);
?>